### PR TITLE
fix(iteration): add type ahead dropdown in iteration association modal

### PR DESCRIPTION
**Pull request by vikram-raj**

### CODE REVIEW

**Comment by fabric8cd** [comment]
@vikram-raj snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1744-fabric8-planner.int.rdu2c.fabric8.io

File: src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html [hunk 1]
```diff
@@ -18,25 +25,25 @@
...
+          <div class="parent-iteration-container input-group dropdown open active">
+            <input id="parent-iteration" class="form-control " type="text" #iterationSearch (keyup)="filterIteration($event)"
+            autocomplete="off" [value]="selectedIterationName" (focus)="iterationSearchFocus();">
+            <ul class="iteration-list dropdown-menu dropdown-ul" *ngIf="showIterationDropdown" #iterationList>
+              <li *ngIf="!filteredIterations.length">No matches found.</li>
+              <li *ngFor="let iteration of filteredIterations;" [attr.data-id]="iteration.key" (click)="associateIteration(iteration);">
+                <span [innerHTML]="(iteration.value) | almSearchHighlight: iterationSearch.value"></span>
+              </li>
+            </ul>
+            <span class="input-group-addon pointer" (click)="iterationSearchFocus();"> <span class="caret"></span> </span>
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html]: this should be a button like the Bootstrap Combobox: http://www.patternfly.org/pattern-library/widgets/#bootstrap-combobox

Comment (mindreeper2420) [inline src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html]: this should be a button like the Bootstrap Combobox: http://www.patternfly.org/pattern-library/widgets/#bootstrap-combobox

File: src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+<modal
+  #iterationAssociationModal
+  title="Associate with Iteration"
+  (onClose)="actionOnClose()"
+  (onOpen)="actionOnOpen()">
   <modal-content>
     <div class="col-md-12 col-sm-12 col-xs-12">
 
       <!-- UX feed for existing iteration association -->
       <div *ngIf="!!workItem.relationships.iteration" id="reassociate-msg">
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html]: this all appears to be clickable, as the cursor changes to the pointer.

File: src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html [hunk 3]
```diff
@@ -18,25 +25,25 @@
...
+              </li>
+            </ul>
+            <span class="input-group-addon pointer" (click)="iterationSearchFocus();"> <span class="caret"></span> </span>
           </div>
         </div>
       </div>
 
       <!-- Action buttons to commit/abort -->
       <div class="text-right">
         <button class="btn" type="submit" id="cancel-iteration-button" (click)="iterationAssociationModal.close()">Cancel</button>
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html]: this should be `<button class="btn btn-default">` and should not need custom styling

http://www.patternfly.org/pattern-library/widgets/#buttons

File: src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.scss [hunk 4]
```diff
@@ -3,3 +3,20 @@
 #cancel-iteration-button {
   margin-right: em(10);
 }
+
+.iteration-list {
+  width: $width100;
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.scss]: no need to use a variable - just use `width: 100%;`

File: src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html [hunk 5]
```diff
@@ -18,25 +25,25 @@
...
-                  <a role="menuitem" tabindex="-1" (click)="onChangeIteration(iteration)">
-                    {{ iteration.attributes.name }}
-                  </a>
-                </li>
-              </ul>
+          <div class="parent-iteration-container input-group dropdown open active">
+            <input id="parent-iteration" class="form-control " type="text" #iterationSearch (keyup)="filterIteration($event)"
+            autocomplete="off" [value]="selectedIterationName" (focus)="iterationSearchFocus();">
+            <ul class="iteration-list dropdown-menu dropdown-ul" *ngIf="showIterationDropdown" #iterationList>
+              <li *ngIf="!filteredIterations.length">No matches found.</li>
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html]: Typeahead dropdown needs to be reset before closing.
1. Open the iteration association modal
2. In the type-ahead, enter a string which results in "No matches found"
3. Close the modal
4. Open the modal - the type-ahead shows No matches found.

File: src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html (hunk 1 above)
Comment (vikram-raj) [inline src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html]: @mindreeper2420 Patternfly Combobox also uses span.

**Comment by fabric8cd** [comment]
@vikram-raj snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1744-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@vikram-raj snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1744-fabric8-planner.int.rdu2c.fabric8.io

**Comment by michaelkleinhenz** [comment]
[test]

**Comment by michaelkleinhenz** [comment]
[test]

**Comment by fabric8cd** [comment]
@vikram-raj snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1744-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@vikram-raj snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1744-fabric8-planner.int.rdu2c.fabric8.io

**Pull request merged**